src/clients/ProfitClient.ts

```diff
@@ -116,32 +153,50 @@ export class ProfitClient {
   }
 
   async update() {
+    // Short circuit early if profitability is disabled.
+    if (!this.enableProfitability) return;
+
     const l1Tokens: { [k: string]: L1Token } = Object.fromEntries(
       this.hubPoolClient.getL1Tokens().map((token) => [token["address"], token])
     );
 
     this.logger.debug({ at: "ProfitClient", message: "Updating Profit client", l1Tokens });
-    let cgPrices: Array<CoinGeckoPrice> = [];
+    let cgPrices: CoinGeckoPrice[] = [];
     try {
-      cgPrices = await this.coingeckoPrices(Object.keys(l1Tokens));
+      const [maticTokenPrice, otherTokenPrices] = await Promise.all([
+        // Add WMATIC for gas cost calculations.
+        this.coingeckoPrices([WMATIC], "polygon-pos"),
+        this.coingeckoPrices(Object.keys(l1Tokens)),
+      ]);
+      cgPrices = cgPrices.concat(maticTokenPrice);
+      cgPrices = cgPrices.concat(otherTokenPrices);
     } catch (err) {
       this.logger.warn({ at: "ProfitClient", message: "Failed to retrieve prices.", err, l1Tokens });
     }
 
+    // Add to l1Tokens after the fetches, so prices and l1Tokens have the same entries, for any error logging later.
+    l1Tokens[WMATIC] = {
+      address: WMATIC,
+      symbol: "WMATIC",
+      decimals: 18,
+    };
+
     const errors: Array<{ [k: string]: string }> = [];
     for (const address of Object.keys(l1Tokens)) {
       const tokenPrice: CoinGeckoPrice = cgPrices.find(
         (price) => address.toLowerCase() === price.address.toLowerCase()
       );
 
-      // todo: Any additional validation to do? Ensure that timestamps are always moving forwards?
-      if (tokenPrice === undefined || typeof tokenPrice.price !== "number") {
+      // TODO: Any additional validation to do? Ensure that timestamps are always moving forwards?
+      if (tokenPrice !== undefined && typeof tokenPrice.price === "number") {
```

Comment:
```suggestion
      if (tokenPrice !== undefined && !isNan(tokenPrice.price)) {
```

nit: not your change but these `typeof`s that paul left in we should try to remove. I don't think it matters that much if the `.price` is a string or number, as long as it can be coerced into a number by `toBNWei(string | number)`